src/jvmMain/kotlin/com/charleskorn/kaml/Yaml.kt

```diff
@@ -49,15 +49,24 @@ public actual class Yaml(
     }
 
     private fun <T> decodeFromReader(deserializer: DeserializationStrategy<T>, source: Reader): T {
-        val parser = YamlParser(source)
-        val reader = YamlNodeReader(parser, configuration.extensionDefinitionPrefix)
-        val rootNode = reader.read()
-        parser.ensureEndOfStreamReached()
+        val rootNode = parseToYamlNodeFromReader(source)
 
         val input = YamlInput.createFor(rootNode, serializersModule, configuration, deserializer.descriptor)
         return input.decodeSerializableValue(deserializer)
     }
 
+    public fun parseToYamlNode(string: String): YamlNode = parseToYamlNodeFromReader(StringReader(string))
+
+    public fun parseToYamlNode(source: InputStream): YamlNode = parseToYamlNodeFromReader(InputStreamReader(source))
+
+    private fun parseToYamlNodeFromReader(source: Reader): YamlNode {
```

Comment:
It would be nice to continue the pattern of all parsing methods using the verb "decode" in their names - so these methods could be something like `decodeToYamlNode`. Or did you deliberately choose to use "parse"?

Reply:
This was indeed a deliberate decision to mirror [parseToJsonElement](https://kotlin.github.io/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json/parse-to-json-element.html) from kotlinx.serialization. I did think about naming them `decode...` but it seemed to me like being closer to kotlinx.serialization.json is more important.

Looking at it more closely now, it stems from the fact that encoding/decoding happens with `@Serializable` classes, and here no `@Serializable` classes are involved. Their [documentation](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/json.md#parsing-to-json-element) puts it well:

> A string can be parsed into an instance of [JsonElement](https://kotlin.github.io/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-element/index.html) with the [Json.parseToJsonElement](https://kotlin.github.io/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json/parse-to-json-element.html) function. It is called neither decoding nor deserialization because none of that happens in the process. It just parses a JSON and forms an object representing it

Here my suggestion is to take the same path and reserve both `decodeFromYamlNode` and `encodeToYamlNode` for `YamlNode` <-> `@Serializable` conversion.

Reply:
Makes sense to me - thanks for clarifying.